### DISCUSSION TITLE
Linux keyring issues

### DISCUSSION THREAD

**Discussion by DominiqueFrt** [borgbase/vorta#859] [General]
With Kubuntu 20.04 - Vorta 0.7.2 - Borg 1.1.15
Problem since the update at the end of January
Identical problem on 2 machines with 3 accounts each
Complete reinstallation from PPA then Flatpack without success

`2021-02-09 13:05:51,922 - vorta.i18n - DEBUG - Loading translation succeeded for ['fr', 'fr-FR', 'fr-Latn-FR'].
2021-02-09 13:05:51,948 - apscheduler.scheduler - INFO - Scheduler started
2021-02-09 13:05:52,195 - root - INFO - Using NetworkManagerMonitor NetworkStatusMonitor implementation.
2021-02-09 13:05:52,202 - vorta.network_status.network_manager - WARNING - Couldn't load settings for /org/freedesktop/NetworkManager/Settings/1
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 52, in get_known_wifis
    settings = self._nm.get_settings(connection_path)
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 139, in get_settings
    return get_result(settings.call('GetSettings'))
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 163, in get_result
    raise DBusException("DBus call failed: {}".format(msg.arguments()))
vorta.network_status.network_manager.DBusException: DBus call failed: ['uid 1002 has no permission to perform this operation']
2021-02-09 13:05:52,277 - vorta.borg.borg_thread - INFO - Running command /app/bin/borg --version
2021-02-09 13:06:08,442 - vorta.network_status.network_manager - WARNING - Couldn't load settings for /org/freedesktop/NetworkManager/Settings/1
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 52, in get_known_wifis
    settings = self._nm.get_settings(connection_path)
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 139, in get_settings
    return get_result(settings.call('GetSettings'))
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 163, in get_result
    raise DBusException("DBus call failed: {}".format(msg.arguments()))
vorta.network_status.network_manager.DBusException: DBus call failed: ['uid 1002 has no permission to perform this operation']
2021-02-09 13:06:12,092 - asyncio - DEBUG - Using selector: EpollSelector
2021-02-09 13:06:48,008 - vorta.borg.borg_thread - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.
2021-02-09 13:06:54,450 - vorta.network_status.network_manager - WARNING - Couldn't load settings for /org/freedesktop/NetworkManager/Settings/1
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 52, in get_known_wifis
    settings = self._nm.get_settings(connection_path)
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 139, in get_settings
    return get_result(settings.call('GetSettings'))
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 163, in get_result
    raise DBusException("DBus call failed: {}".format(msg.arguments()))
vorta.network_status.network_manager.DBusException: DBus call failed: ['uid 1002 has no permission to perform this operation']
2021-02-09 13:06:54,479 - vorta.network_status.network_manager - WARNING - Couldn't load settings for /org/freedesktop/NetworkManager/Settings/1
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 52, in get_known_wifis
    settings = self._nm.get_settings(connection_path)
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 139, in get_settings
    return get_result(settings.call('GetSettings'))
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 163, in get_result
    raise DBusException("DBus call failed: {}".format(msg.arguments()))
vorta.network_status.network_manager.DBusException: DBus call failed: ['uid 1002 has no permission to perform this operation']
2021-02-09 13:08:12,743 - vorta.network_status.network_manager - WARNING - Couldn't load settings for /org/freedesktop/NetworkManager/Settings/1
Traceback (most recent call last):
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 52, in get_known_wifis
    settings = self._nm.get_settings(connection_path)
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 139, in get_settings
    return get_result(settings.call('GetSettings'))
  File "/app/lib/python3.7/site-packages/vorta/network_status/network_manager.py", line 163, in get_result
    raise DBusException("DBus call failed: {}".format(msg.arguments()))
vorta.network_status.network_manager.DBusException: DBus call failed: ['uid 1002 has no permission to perform this operation']
2021-02-09 13:08:47,745 - vorta.borg.borg_thread - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.
2021-02-09 14:05:56,615 - vorta.borg.borg_thread - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.`

cordially

**Comment by m3nu** [comment]
So what's the issue you are reporting and can you try to add your logs with line breaks preserved? Will make it easier to read. Thanks!

**Comment by DominiqueFrt** [comment]
[vorta.log](https://github.com/borgbase/vorta/files/5957575/vorta.log)
Problem: Backup failure.Cannot access an existing repository or create a new repository.
Message: `Please unlock your password manager`
cordially

**Comment by m3nu** [comment]
Have you checked it your issue still happens on the current `master` branch? There is already a fix for a similar issue.

Else one of our Linux maintainers needs to have a look.

**Comment by jetlag666** [comment]
Hello, I also have this issue on Kubuntu 20.04 since 3 days ago.

**Comment by m3nu** [comment]
Please try the current `master` branch to verify if the fix (#779) applied there resolves your issue.

**Comment by peerwal** [comment]
Hi, i am running into the same issue using KDE Neon 5.20. Both the flatpak and the latest deb package install come up with the same message: **Please unlock your password manager**

**Comment by DominiqueFrt** [comment]
I purged Vorta and Borg
I deleted :
  ~ / .cache / borg /
  ~ / .cache / vorta /
  ~ / .config / borg /
  ~ / .config / vortac
  ~ / .local / share / vorta /
  ~ / .var / app / com.borgbase.Vorta /

Reinstallation from your site: https://vorta.borgbase.com
/ install / linux /

Creation of a new depot

Flatpack: Please unlock your password manager
PPA: Same message

Are there other folders to delete to reset everything

cordially

**Comment by m3nu** [comment]
Sorry, but I'm unable to reproduce this on Gnome (Fedora/SecretStorage keyring) or KDE (Kubuntu/KWallet keyring) using the latest `master` branch installed via `pip`.

I can't comment on individual distribution packages, since they are maintained by external packagers listed here: https://vorta.borgbase.com/install/linux/

**Comment by DominiqueFrt** [comment]
For information

![Screenshot_20210215_120044](https://user-images.githubusercontent.com/42612895/107938365-8fdeaa00-6f85-11eb-8291-d66376606e39.png)
![Screenshot_20210215_114834](https://user-images.githubusercontent.com/42612895/107938373-93723100-6f85-11eb-99c2-3c7b0affb37c.png)
![Screenshot_20210215_120400](https://user-images.githubusercontent.com/42612895/107938664-011e5d00-6f86-11eb-917b-1417179e491a.png)

Cordially

**Comment by DominiqueFrt** [comment]
After restarting PC:
  Administrator account: OK
  User account: Problem

![administrator account](https://user-images.githubusercontent.com/42612895/107947900-5f057180-6f93-11eb-86c1-ed735d28ac93.png)
![user account](https://user-images.githubusercontent.com/42612895/107947910-62006200-6f93-11eb-9910-a55660dabf0e.png)

Hope this can help

**Reply by m3nu** [reply]
The issue you are facing is (most likely) already fixed in Github's `master` branch. So either install that or wait for the next 0.7.3 release. Based on your screenshot, you installed version 0.7.2 from PyPi. If you are unsure on how to install the current `master` branch, please see our [developer docs](https://vorta.borgbase.com/contributing/#local-development-setup).

The logs in your screenshot are unrelated to the keyring.

**Comment by DominiqueFrt** [comment]
For information. Return Vorta 0.7.3
![Vorta 0 7 3](https://user-images.githubusercontent.com/42612895/108061280-5e82de80-7058-11eb-815c-b0195020cb75.png)
[Vorta 0.7.3.pdf](https://github.com/borgbase/vorta/files/5988428/Vorta.0.7.3.pdf)

**Comment by carobinario** [comment]
If you have troubles you should give [KeepassXC](https://keepassxc.org/) a try: it's a password manager that supports FreeDesktop.org Secret Service standard, so it can store password from apps, like Vorta, that use that standard.

I had the same _Please unlock your password manager_ issue in Fedora Kde spin. I removed gnome-keyring, disabled kdewallet and enabled Secret Service integration in KeepassXC. Here you can read a tutorial: https://c3pb.de/blog/keepassxc-secrets-service.html. 

_Tested with Vorta 0.7.2 installed via Flathub._

**Comment by m3nu** [comment]
Is your issue resolved with the latest release 0.7.3? (Since nobody is testing the Github version, we'll just do more frequent releases.)

**Reply by peerwal** [reply]
Yes, thanks for your support. It works fine now. Obviously it didn't work with Kwallet and as long as the gnome-keyring daemon was enabled (wondering, why it obviously is enabled by default even on systems using KDE Plasma). 

I have been using KeepassXC as my default password manager for a long time just had to upgrade it to the latest version 2.6.4 and enable the Secret Servive option to make Vorta use KeepassXC without issueing that annaoying message

**Comment by DominiqueFrt** [comment]
Following the update to Vorta 0.7.3, I stay on Vorta 0.7.2.
Update from the software then with pip3 without success.
How to remove all traces of Vorta to redo a clean installation?
![Screenshot_20210219_111719](https://user-images.githubusercontent.com/42612895/108499156-2d9be700-72ae-11eb-8cf6-c12771cda265.png)
![Screenshot_20210219_121814](https://user-images.githubusercontent.com/42612895/108499167-3096d780-72ae-11eb-8e31-39e00240756a.png)
[Vorta 2021-02-19.pdf](https://github.com/borgbase/vorta/files/6009613/Vorta.2021-02-19.pdf)

**Reply by M4g3dd0** [reply]
I have encountered the same issue. I installed Seahorse and unlocked the password manager. I am on Debian testing, Seahorse version 3.38.0.1. Vorta version 0.7.2 Cinnamon version 4.8.6

**Reply by m3nu** [reply]
> Vorta version 0.7.2

You probably want to update Vorta first. Your issue may be adressed in the latest release.

We are also looking for Linux users to help test the current Github version *before* it's released. Else it's unavoidable that some bugs will be released on Linux.

**Comment by DominiqueFrt** [comment]
Installation Vorta 0.7.3 from GitHub source, if I have correctly interpreted the instructions
[Vorta 0.7.3.pdf](https://github.com/borgbase/vorta/files/6014781/Vorta.0.7.3.pdf)

**Reply by m3nu** [reply]
Nice graphic design work! 👌

To upgrade a package with Pip, try: `$ pip install --upgrade vorta`

Regarding your error, I don't think it's related to Vorta. Your DBus + Kwallet has some issue. Sadly I can't tell you more than that.

If you have time to debug it yourself, you can follow the code in [vorta/keyring/kwallet.py](https://github.com/borgbase/vorta/blob/master/src/vorta/keyring/kwallet.py). In `try_unlock()` it would usually give you a long number. In your case you get an error message. So KWallet is there in `__init__()`, but causes an error later. I can't reproduce this (Kubuntu 20.20), but if we get more such reports, we should probably unlock the repo in `VortaKWallet5Keyring.__init__()` and handle the error there.

![Screen Shot 2021-02-20 at 20 06 34](https://user-images.githubusercontent.com/3916435/108595293-10543f00-73ba-11eb-91fc-1fcce9c7d7ed.jpg)

**Comment by JanSchnacki** [comment]
I just updated vorta to 0.7.3 via the Ubuntu PPA package, and... YAY! I can backup again.

KWallet is detected successfully again.

**Comment by DominiqueFrt** [comment]
Vorta 0.7.3 is finally installed
I ended up with 2 different vorta versions on 2 accounts.
So I placed the command "pip3 uninstall vorta" from each account and reinstall vorta from the PPA
The situation is the same with Vorta 0.7.3, works with KeePassXC

I submitted a request to upgrade KeePassXC
[](https://github.com/keepassxreboot/keepassxc/issues/6157)

I installed d-feet present in the Discover software library.
I don't have the skill to use this tool. Now I can make "gnome-keyring-daemon" and "kwalletd5" executable, restart Vorta without KeePassXC, and return the d-feet information to you if you guide me.

**Comment by neurer** [comment]
Please forgive if it's not the right place to ask, but my issue appears to be related. I'm on 0.7.3 via samuel-w1 PPA. Backups stopped working with the last update as it asks for KDE Wallet Service. I'm using Gnome keyring/Seahorse only.

**Reply by m3nu** [reply]
Hi and thanks for reporting this.

In order to open a bug issue for Vorta, please help me reproduce the issue first. What steps can I follow on Gnome to see the issue? I'm using Fedora 33 for my Gnome test desktop.

**Reply by neurer** [reply]
Thanks for the quick reply. Hope this helps:

* run Vorta 0.7.2 in Gnome environment; Ubuntu 20.04 here
* have two local machines talking to each other via ssh
* ssh key is password protected
* save password to Gnome keyring
* update to Vorta 0.7.3

Same day scheduled backups stopped working, apt logs show _Unpacking vorta (0.7.3-0ubuntu0) over (0.7.2-0ubuntu1)_. I do not remember the contents of the dialogue window, but it said something like securing with KWallet and gives you two encryption options to chose from. It appears as if the previously working Gnome keyring is skipped entirely.

**Reply by m3nu** [reply]
SSH password protection isn't handled by Vorta.

There were a few [changes](https://github.com/borgbase/vorta/blame/master/src/vorta/keyring/abc.py) to keyring handling in the last update. Not an expert myself for Linux keyrings. I can only suggest to re-add any broken repos, so the password ends up in the right place, if it wasn't before.

Any idea on how to resolve these new issues, @samuel-w ? Maybe we can add a update note?

**Reply by samuel-w** [reply]
The only keyring changes were swapping kwallet and secretservice priorities, so it goes kwallet -> secretservice -> db.

**Reply by samuel-w** [reply]
Just tested, Ubuntu 20.04 GNOME desktop and it uses secretservice keyring fine.

**Reply by neurer** [reply]
![Screenshot from 2021-02-24 17-56-17](https://user-images.githubusercontent.com/6098965/109036197-a5607c00-76c9-11eb-97e2-a57e6885b86b.png)


```
apt-cache policy "*kwallet*"
kwalletmanager:
  Installed: (none)
  Candidate: 4:19.12.3-0ubuntu1
  Version table:
     4:19.12.3-0ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
libkwalletbackend5-5:
  Installed: 5.68.0-0ubuntu2
  Candidate: 5.68.0-0ubuntu2
  Version table:
 *** 5.68.0-0ubuntu2 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
        100 /var/lib/dpkg/status
libpam-kwallet-common:
  Installed: (none)
  Candidate: 4:5.18.4.1-0ubuntu1
  Version table:
     4:5.18.4.1-0ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
        500 http://de.archive.ubuntu.com/ubuntu focal/universe i386 Packages
kwalletcli:
  Installed: (none)
  Candidate: 3.02-1build1
  Version table:
     3.02-1build1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
signon-kwallet-extension:
  Installed: (none)
  Candidate: 4:19.12.3-0ubuntu1
  Version table:
     4:19.12.3-0ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
remmina-plugin-kwallet:
  Installed: (none)
  Candidate: 1.4.2+dfsg-1ubuntu1
  Version table:
     1.4.2+dfsg-1ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
libpam-kwallet4:
  Installed: (none)
  Candidate: 4:5.18.4.1-0ubuntu1
  Version table:
     4:5.18.4.1-0ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
libpam-kwallet5:
  Installed: (none)
  Candidate: 4:5.18.4.1-0ubuntu1
  Version table:
     4:5.18.4.1-0ubuntu1 500
        500 http://de.archive.ubuntu.com/ubuntu focal/universe amd64 Packages
```

**Reply by neurer** [reply]
Updated to 0.7.4 and am still seeing the KWallet prompt; scheduled backups are not working. Any suggestions would be appreciated.

**Reply by m3nu** [reply]
If you don't use KWallet, you can probably uninstall it?

For now we only plan to give the option of using the available system keychain or Vorta's internal settings database. Please test the current pull request and let us know if anything could be improved with this new feature: #898

**Reply by mauromol** [reply]
Just my 2cents... I think there's a fundamental misconception here. The problem is **not** that we don't want to use KWallet rather than Vorta's internal keyring, the problem is that, if I understood it right, Vorta from version 0.7.3 started to prefer the system keyring over the internal one. For this reasons, passwords that were previously saved into the internal keyring are now searched in KWallet first and, since they're not there, backups won't work any more.
If you unlink and re-add your repositories, they will be back functional, because Vorta will now store passwords in KWallet and then retrieve them from it on subsequent requests.

So, the actual problem introduced in Vorta 0.7.3 is indeed a breaking change that will cause previous installations to not find keys previously saved in the internal keyring any more. So, the solution is either:
- document it in some sort of "upgrade notes" and force users to unlink and re-add their repositories
- provide a conversion process on first start after the upgrade (or using the package upgrade scripts) to retrieve repository passwords from the internal keyring and move them into KWallet (if present) so that subsequent Vorta backups will succeed to find them in KWallet
- provide a fallback mechanism within Vorta that, if the password was not found in KWallet, it is searched in the internal keyring (while, in my experience, the original 0.7.3 just failed to run backups complaining that no password could be found, after just searching into KWallet)

This is just IMHO. Suggesting to uninstall KWallet, in any case, cannot be a solution.

**Reply by neurer** [reply]
Well, my config worked fine with Gnome keyring up until the 0.7.3 update. I did see, thanks for pointing it out, that keyring order has been changed. I get the use case. But my problem is, that it doesn't seem like KWallet is installed. At least not wittingly. The only KWallet related package i'm able to find, is "libkwalletbackend5-5" as per log posted above. Anything else i should be looking for?

**Reply by m3nu** [reply]
This fallback mechanism already exists. It will check in the system keychain, as well as the internal settings: https://github.com/borgbase/vorta/blob/master/src/vorta/borg/borg_thread.py#L145

Maybe you have some ideas on how to solve this properly? I'm not a Linux desktop user and lack the knowledge on what's common.

**Reply by m3nu** [reply]
> The only KWallet related package i'm able to find, is "libkwalletbackend5-5" as per log posted above. Anything else i should be looking for?

It will prefer KWallet because @samuel-w , who implemented the feature noticed that Gnome Keyring is often installed, but not used. Not sure if that's the case. Code here: https://github.com/borgbase/vorta/blob/master/src/vorta/keyring/abc.py#L30

That's all. If there is a better solution, I'm open to pull requests.

**Reply by mauromol** [reply]
Perhaps in your case Vorta is now detecting the presence of KWallet even if you're probably not using it, just because you have the backend installed (possibly because it was automatically installed by some KDE program you use). The final effect is the same, the changed order with which passwords are searched for has broken your install. Maybe in your case a better way to detect what is the actual default system keyring would fix the problem, although I still think that the change introduced in Vorta 0.7.3 would require a migration process or a fallback mechanism in password search algorithm.

**Reply by m3nu** [reply]
python-keyring looks at the current desktop. Maybe we use that to avoid the issue described above? https://github.com/jaraco/keyring/blob/cca9efce749b273b453573d047b887382d8c6896/keyring/backends/kwallet.py#L56

**Reply by mauromol** [reply]
@m3nu sorry, I don't know Python and the internals of this, but as I described in #871, such fallback mechanism seemed to be broken in some way. True, the logs said "Password not found in primary keyring. Falling back to VortaDBKeyring", but for some reason this was not working and I can state for sure that I just did an upgrade from 0.7.2->0.7.3 without doing any change at all to my configuration, but backups stopped working.

neurer seems in a similar position: Vorta is now searching for passwords in KWallet first, if it can't find them there it should find them in Gnome keyring but it seems like it's failing for some reason. Unfortunately I can't be of further help, also because I unlinked and readded my repositories and now my Vorta is back functional, so I can't even test to see if a new version fixes this.

**Reply by m3nu** [reply]
Fair enough. Let's use KWallet only on KDE and see what feedback comes back.

**Reply by neurer** [reply]
@mauromol Yes, that's what it looks like to me too. Here's what removing _libkwalletbackend5-5_ would do:

```
sudo apt remove libkwalletbackend5-5
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  kwayland-data kwayland-integration libdbusmenu-qt5-2 libfam0 libgpgmepp6 libkf5auth-data libkf5authcore5 libkf5codecs-data libkf5codecs5 libkf5config-bin libkf5config-data libkf5configcore5 libkf5configgui5 libkf5configwidgets-data libkf5configwidgets5 libkf5coreaddons-data
  libkf5coreaddons5 libkf5crash5 libkf5dbusaddons-bin libkf5dbusaddons-data libkf5dbusaddons5 libkf5guiaddons5 libkf5i18n-data libkf5i18n5 libkf5idletime5 libkf5notifications-data libkf5notifications5 libkf5service-bin libkf5service-data libkf5service5 libkf5wallet-data
  libkf5waylandclient5 libkf5widgetsaddons-data libkf5widgetsaddons5 libkf5windowsystem-data libkf5windowsystem5 libpolkit-qt5-1-1 libqt5texttospeech5
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  libkf5wallet-bin libkf5wallet5 libkwalletbackend5-5
```
Not exactly sure, but these appear to be the remains of a Kdenlive install. I'll have another look in the next days. 

Never the less, a different approach to selecting keyrings seems worth considering.

**Reply by m3nu** [reply]
Based on the feedback here, it seems that KWallet is also commonly installed. Same as Secret Storage.

So I'll do the same as python-keyring and use KWallet only if the desktop is KDE. This works as expected in the VMs I test (Kubuntu/KDE, Fedora/Gnome, Debian 11/KDE).

If you can, please help test the pull request [here](https://github.com/borgbase/vorta/pull/898) or view the [new rules to select the correct keyring](https://github.com/borgbase/vorta/pull/898/files#diff-93338d80b527a6798bcb9c791af2eba61822082bc1f605e33273d9084ae5f823R36).

**Reply by neurer** [reply]
Went ahead and did some testing with 0.7.4:

* removed _libkwalletbackend5-5_ and its dependencies: Vorta uses Gnome keyring fine
* re-installed _libkwalletbackend5-5_ and its dependencies: Vorta uses Gnome keyring fine
* _libkwalletbackend5-5_ does **not** pull _libkf5wallet-bin_ as a dependency. Went ahead and installed _libkf5wallet-bin_: Vorta tries using KWallet and fails

It seems as if having _libkf5wallet-bin_ and it's dependencies installed will interfere with selecting the correct keyring. Please mind that this still doesn't mean KWallet is usable in a (non-KDE) desktop environment. So any app that requires _libkf5wallet-bin_ to be present would lead to Vorta keyring selection issues.

Hope this helps with testing any proposed changes.

Last but not least. For now, with _libkf5wallet-bin_ and dependencies gone, my backups are working again. Without config changes or re-linking.

**Reply by mauromol** [reply]
So, if I understand it right, if you **don't** have `libkf5wallet-bin` installed, backups are working fine. If it is, Vorta tries to use KWallet, it can't find the passwords there and backups fail.
If this is the case, this is another clue of the fact that the fallback mechanism is not working properly (in addition to the fact that the use of KWallet is unexpected in your case).

**Reply by m3nu** [reply]
The fallback is from the detected system keychain to Vorta’s DB. It’s **not** to another system keychain.

**Comment by DominiqueFrt** [comment]
Bonjour,

A ce jour Vorta 0.7.3 fonctionne :
    Avec KeePassXC sur mon compte administrateur
    Sans KeePassXC sur mes 2 comptes utilisateur
    
Je ne comprend pas ce qui fais la différence.
Cependant, le fait que Vorta fonctionne différemment sur 2 comptes peut permettre de comparer des fichiers.
Je peux poster les fichiers qui pourraient être utiles.

Sur ma seconde machine je n'ai pas eu de problème avec Vorta 0.7.3 sauf devoir reconfigurer les dépôts.

Cordialement
________________________________________________________________________________________
Hi,

To date Vorta 0.7.3 works:
    With KeePassXC on my administrator account
    Without KeePassXC on my 2 user accounts
    
I don't understand what makes the difference.
However, the fact that Vorta works differently on 2 accounts may allow you to compare files.

I can post any files that might be useful.

On my second machine I had no problem with Vorta 0.7.3 except having to reconfigure the repositories.

Cordially

[⁄usr⁄lib⁄python3⁄dist-packages⁄vorta⁄keyring 
.pdf](https://github.com/borgbase/vorta/files/6029706/usr.lib.python3.dist-packages.vorta.keyring.pdf)
[Compte administrateur | ~⁄.cache⁄log⁄vorta.pdf](https://github.com/borgbase/vorta/files/6029707/Compte.administrateur.cache.log.vorta.pdf)
[Compte utilisateur | ~⁄.cache⁄log⁄vorta.pdf](https://github.com/borgbase/vorta/files/6029708/Compte.utilisateur.cache.log.vorta.pdf)

**Reply by m3nu** [reply]
For your admin account, SecretStorage seems to work, but the password still comes from Vorta's DB. Maybe SS didn't work before?

For the user accounts, they don't seem to have access to a keyring and Vorta's DB is used.

In general, we first try to use a system keyring (KWallet or SS) and then fall back to keeping the password in the DB.

Also recommend to send logs as `.log`, as you find them instead PDF. A bit hard to read.

**Reply by samuel-w** [reply]
Looks like your dbus permissions are broken, judging by this error message for the NetworkManager DBus.
```DBus call failed: ['uid 1000has no permission to perform this operation']```

**Reply by m3nu** [reply]
> Looks like your dbus permissions are broken, judging by this error message for the NetworkManager DBus.
`DBus call failed: ['uid 1000has no permission to perform this operation']`

Careful. This error could be from `network_manager`. It tries getting the current and past Wifis in a few different ways and prints the error if it fails.

For everyone else having this error: There is now a potential [fix](https://github.com/borgbase/vorta/pull/889) merged into the `master` branch. You can try it using

`$ pip install --force-reinstall git+https://github.com/borgbase/vorta.git`

**Reply by samuel-w** [reply]
I also patched it in to my PPA.

**Comment by m3nu** [comment]
If the master branch is working for all Linux users, we should do a small release soon.

Would appreciate feedback on this. Quick way to install from the current `master` branch:

`$ pip install --force-reinstall git+https://github.com/borgbase/vorta.git`

**Comment by m3nu** [comment]
Did a minor release to hopefully address all the issues reported here. Thanks to everyone for adding fixes so quickly.

If you want to make sure the next release works well on your platform and give early feedback, please consider joining the [*Pre-release testing* thread](https://github.com/borgbase/vorta/discussions/895).

**Comment by m3nu** [comment]
Maybe we should make the keyring integration optional. So Linux users can disable it, if they don't use the keyring that's installed.: https://github.com/borgbase/vorta/pull/898

**Reply by samuel-w** [reply]
I don't see the case where I have a password manager installed and I don't want to use it in applications. I'd rather fix/remove the code causing the bugs.

**Comment by DominiqueFrt** [comment]
Hi,

Test today Vorta 0.7.4
Passwords with Kwallet or KeePassXC: OK
One minor note: The main window opens at startup.

![Screenshot_20210227_105354](https://user-images.githubusercontent.com/42612895/109384293-1c755a80-78ec-11eb-9294-bbbba74b31a5.png)

**Comment by m3nu** [comment]
[Minor release 0.7.5](https://github.com/borgbase/vorta/releases/tag/v0.7.5) is out that addresses the keyring issues reported here. Main changes:

- You can disable the system keychain and just keep the repo passphrase in Vorta's settings DB (slightly less secure)
- Keyring is now picked by assigning a score. So it will prioritized KWallet on KDE and Secret Storage on Gnome.

Maybe additional rules could be added here. If you do NOT use Gnome or KDE, but have both, KWallet and SecretStorage installed, it will use SecretStorage.

**Reply by neurer** [reply]
Thanks for staying on top of this issue. I thought I'd help test the new release but it appears that `ppa:samuel-w1/vorta` has been removed from https://vorta.borgbase.com/install/linux/ and 0.7.5 isn't available yet. Any pointers would be appreciated.

**Reply by m3nu** [reply]
There is now an official Debian package, maintained by Nicholas Steeves: https://packages.debian.org/unstable/utils/vorta

Already available for Debian 11. Will eventually move to derivatives, like Ubuntu. Since Vorta is all Python code, it may work on older versions when just installing the .deb. (untested)

Samuel retired his PPA and isn’t updating it any more.

As another alternative, using Flatpak or Pip with our [.desktop](https://github.com/borgbase/vorta/blob/master/src/vorta/assets/metadata/com.borgbase.Vorta.desktop) file should also work well.

**Reply by neurer** [reply]
Went ahead and manually installed `vorta_0.7.5-1_all.deb` from https://packages.debian.org/unstable/utils/vorta over 0.7.4 from `ppa:samuel-w1/vorta`. I can confirm that this release fixes the keyring issues even with [Kwallet dependencies](https://github.com/borgbase/vorta/discussions/859#discussioncomment-415527) present. Excellent work.

To address the valid expectation of receiving updates and avoid additional confusion, might i suggest adding a footnote to https://vorta.borgbase.com/install/linux/ that `ppa:samuel-w1/vorta` is unmaintained.

**Reply by m3nu** [reply]
Actually it was fully removed 2 days ago. Do you see a an older, cached version on your end?

https://github.com/borgbase/vorta.borgbase.com/pull/21

**Reply by neurer** [reply]
No i do not see a cached version; i mentioned above that it's missing from the page. But thinking of people that are currently using the PPA, it might be good to communicate that it's now unmaintained. If someone has previously added this PPA, that person has an interest in updated packages as well as a valid expectation of further updates.

It's just a suggestion. Feel free to ignore.

**Reply by m3nu** [reply]
Ah. I see. Actually the PPA itself already says just that. https://launchpad.net/~samuel-w1/+archive/ubuntu/vorta

**Reply by mauromol** [reply]
> As another alternative, using Flatpak or Pip with our [.desktop](https://github.com/borgbase/vorta/blob/master/src/vorta/assets/metadata/com.borgbase.Vorta.desktop) file should also work well.

By the way, is there a reason why the Flatpak version does not install the desktop file itself? All the other applications I ever installed from Flathub do install an icon in the application menù, unless of course they are command-line applications. Vorta does not.

**Reply by m3nu** [reply]
The Flatpak install script definitely uses the .desktop file: https://github.com/borgbase/vorta/blob/master/Makefile#L58

Not sure why it wouldn't work in your case. We can ping @Hofer-Julian. He's the Flatpak expert. 🤓

**Reply by Hofer-Julian** [reply]
It should install the desktop file.
It might be that your distro requires you to log out and log on again before it shows up.

**Reply by mauromol** [reply]
That's strange, usually I do not need to logout and login again. I will retry: this morning Flatpack didn't add the Vorta icon in my application menù. Then I uninstalled it mainly because I saw Vorta-Flatpack didn't pick up the settings from the Vorta-DEB version installed through the old PPA (suggestions welcome: current settings seem to be stored in `~/.local/share/Vorta`). I intend to try it again sooner or later, but today I don't have time.

Thank you for clarification.

**Reply by Hofer-Julian** [reply]
See https://vorta.borgbase.com/install/linux/ for copying settings

**Reply by alastairdb** [reply]
I'm running KDE on Ubuntu 20.04. I have two pip installed versions of vorta. Version 0.6.24 works fine. Version 0.7.5 gives me the following error message:

> Your repo passphrase was stored in a password manager which is no longer available.
> Try unlinking and re-adding your repo.

I get this message no matter what is set in "Store repository passwords in system keychain, if available."

I'm reluctant to unlink and relink in case I lose all the information in the "Archives" tab. I guess I'll keep using 0.6.24 until it breaks  or I learn more about relinking

**Reply by m3nu** [reply]
Archive details will be fetched from the repo when re-linking. You will only loose the size of the added data.

I would still recommend to relink, since currently you don't know where your password is saved. Previous Vorta versions assumed only one password manager is installed at a time. This turned out to be too naive, since many system come with both, SecureStorage and KWallet. For your case, I guess the password is in SS, but now Vorta looks in KWallet only. Hence the message you get.

Alternatively you may be able to add some random repo and then edit the URL and password in KWallet to match your other repo. Then it will be found without relinking anything.

**Reply by carobinario** [reply]
Hi,
I'm using **Vorta 0.8.3** (**Borg 1.2.0**) with Opensuse Tumbleweed (20220409), KDE Plasma Version: 5.24.4, KDE Frameworks Version: 5.92.0, Qt Version, 5.15.2. Also I'm using **KeepassXC 2.7** as Secret service password manager.

I always back up without problem to an usb hard disk and to Borgbase (two different profiles).

Today I created a third profile to backup to the usb hard disk. Unfortunately Vorta hanged when I created the repo 

_2022-04-11 18:18:51,027 - vorta.borg.borg_job - WARNING -
IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s)._


After restarting Vorta I tried to backup but I had this error:

_2022-04-11 18:29:15,829 - vorta.borg.borg_job - DEBUG - Password not found in primary keyring. Falling back to VortaDBKeyring.
2022-04-11 18:29:15,830 - vorta.notifications - DEBUG - notification not suppressed_

What's wrong? Why can I backup to the other two repositories but not to the third one?

Here is the log: [vorta.log](https://github.com/borgbase/vorta/files/8466192/vorta.log)

Thanks

**Reply by real-yfprojects** [reply]
Hi @carobinario,

I don't think your comment relates to the post you commented on. This discussion is already has a lot of post. I therefore ask you to open a new issue for your problem.

**Reply by carobinario** [reply]
Ok, thanks.
